dapp/src/hooks/store/useActivities.ts

```diff
@@ -2,5 +2,12 @@ import { selectActivities } from "#/store/wallet"
 import { useAppSelector } from "./useAppSelector"
 
 export default function useActivities() {
-  return useAppSelector(selectActivities)
+  const data = useAppSelector(selectActivities)
+  const status: "pending" | "idle" = data.some(
+    (activity) => activity.status === "pending",
+  )
+    ? "pending"
+    : "idle"
```

Comment:
The term `status` is a bit confusing to me. What I mean is that an activity can have a `status`, here I understand it more as a fetching `status`. Maybe we should change the name to `hasAnyPendingActivity` and return boolean value.


Additionally, I think we can create a separate selector for this data. The [role of selectors](https://redux.js.org/usage/deriving-data-selectors#basic-selector-concepts) is to return data based on that state. So it seems that we can keep this approach.

```
export const selectHasAnyPendingActivity = createSelector(
  (state: RootState) => state.wallet.activities,
  (activities) => activities.some((activity) => activity.status === "pending"),
)
```

Reply:
Ref commit: 18a4238c